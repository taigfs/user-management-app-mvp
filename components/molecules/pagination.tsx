import React from "react";
import { View, TouchableOpacity, Text } from "react-native";

interface PaginationProps {
  currentPage: number;
  totalPages: number;
  onPageChange: (page: number) => void;
}

export const Pagination: React.FC<PaginationProps> = ({
  currentPage,
  totalPages,
  onPageChange,
}) => {
  const pageNumbers = Array.from(
    { length: totalPages },
    (_, index) => index + 1,
  );

  return (
    <View className="flex flex-row justify-center">
      {pageNumbers.map((page) => (
        <TouchableOpacity
          key={page}
          onPress={() => onPageChange(page)}
          className={`mx-1 px-3 py-1 rounded ${
            page === currentPage
              ? "bg-blue-500 border border-blue-600"
              : "bg-black border border-gray-600"
          }`}
        >
          <Text className="text-white">{page}</Text>
        </TouchableOpacity>
      ))}
    </View>
  );
};
