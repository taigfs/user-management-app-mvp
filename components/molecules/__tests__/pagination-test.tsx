import React from "react";
import renderer from "react-test-renderer";

import { Pagination } from "../pagination";

it(`renders correctly`, () => {
  const tree = renderer
    .create(
      <Pagination currentPage={1} totalPages={1} onPageChange={() => {}} />,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
