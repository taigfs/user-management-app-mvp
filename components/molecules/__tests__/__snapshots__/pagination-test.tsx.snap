// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  className="flex flex-row justify-center"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      className="text-white"
    >
      1
    </Text>
  </View>
</View>
`;
